File: widgetssdk/src/main/java/com/glia/widgets/chat/domain/CustomCardTypeUseCase.java
package com.glia.widgets.chat.domain;

import androidx.annotation.Nullable;

import com.glia.widgets.chat.adapter.CustomCardAdapter;

public class CustomCardTypeUseCase {
    @Nullable
    private final CustomCardAdapter adapter;

    public CustomCardTypeUseCase(@Nullable CustomCardAdapter adapter) {
        this.adapter = adapter;
    }

    @Nullable
    public Integer execute(int chatAdapterViewType) {
        if (adapter == null) {
            return null;
        }
        return adapter.getCustomCardViewType(chatAdapterViewType);
    }
}

File: widgetssdk/src/main/java/com/glia/widgets/chat/domain/CustomCardTypeUseCase.kt
package com.glia.widgets.chat.domain

import com.glia.widgets.chat.adapter.CustomCardAdapter

internal class CustomCardTypeUseCase(private val adapter: CustomCardAdapter?) {
    operator fun invoke(chatAdapterViewType: Int): Int? {
        return adapter?.getCustomCardViewType(chatAdapterViewType)
    }
}
